src/transformers/audio_utils.py

```diff
@@ -0,0 +1,357 @@
+
+
+    Args:
+        all_freqs (`np.array` of shape (`nb_frequency_bins`, )):
+            Discrete frequencies used when the STFT was computed.
+        f_pts (`np.array`, of shape (`nb_mel_filters`, )):
+            Coordinates of the middle points of the triangular filters to create.
+
+    Returns:
+        fb (np.array):
+            The filter bank of size (`nb_frequency_bins`, `nb_mel_filters`).
+    """
+    # Adapted from Librosa
+    # calculate the difference between each filter mid point and each stft freq point in hertz
+    f_diff = f_pts[1:] - f_pts[:-1]  # (n_filter + 1)
+    slopes = np.expand_dims(f_pts, 0) - np.expand_dims(all_freqs, 1)  # (nb_frequency_bins, n_filter + 2)
+    # create overlapping triangles
+    zero = np.zeros(1)
+    down_slopes = (-1.0 * slopes[:, :-2]) / f_diff[:-1]  # (nb_frequency_bins, n_filter)
+    up_slopes = slopes[:, 2:] / f_diff[1:]  # (nb_frequency_bins, n_filter)
+    fb = np.maximum(zero, np.minimum(down_slopes, up_slopes))
+
+    return fb
+
+
+def get_mel_filter_banks(
+    nb_frequency_bins: int,
+    nb_mel_filters: int,
+    frequency_min: float,
+    frequency_max: float,
+    sample_rate: int,
+    norm: Optional[str] = None,
+    mel_scale: str = "htk",
+) -> np.array:
+    """
+    Create a frequency bin conversion matrix used to obtain the Mel Spectrogram. This is called a *mel filter bank*,
+    and various implementation exist, which differ in the number of filters, the shape of the filters, the way the
+    filters are spaced, the bandwidth of the filters, and the manner in which the spectrum is warped. The goal of these
+    features is to approximate the non-linear human perception of the variation in pitch with respect to the frequency.
+    This code is heavily inspired from the *torchaudio* implementation, see
+    ![here](https://pytorch.org/audio/stable/transforms.html) for more details.
+
+
+    Note:
+        Different banks of Mel filters were introduced in the litterature. The following variation are supported:
+            - MFCC FB-20: introduced in 1980 by Davis and Mermelstein, it assumes a sampling frequency of 10 kHertz
+                and a speech bandwidth of `[0, 4600]` Hertz
+            - MFCC FB-24 HTK: from the Cambridge HMM Toolkit (HTK) (1995) uses a filter bank of 24 filters for a
+                speech bandwidth `[0, 8000]` Hertz (sampling rate ≥ 16 kHertz).
+            - MFCC FB-40: from the Auditory Toolbox for MATLAB written by Slaney in 1998, assumes a sampling rate
+                of 16 kHertz, and speech bandwidth [133, 6854] Hertz. This version also includes an area normalization.
+            - HFCC-E FB-29 (Human Factor Cepstral Coefficients) of Skowronski and Harris (2004), assumes sampling
+                rate of 12.5 kHertz and speech bandwidth [0, 6250] Hertz
+        The default parameters of `torchaudio`'s mel filterbanks implement the `"htk"` filers while `torchlibrosa` uses
+        the `"slaney"` implementation.
```

Comment:
Consider using the Tip syntax, also indentation in list items needs fixing (second line after the list item should be indented with the text above in Markdown syntax).